src/utils/arrays.ts

```diff
@@ -0,0 +1,23 @@
+export const equalArrays = (array1?: unknown[] | null, array2?: unknown[] | null): boolean => {
+  if (array1 && !array2) {
+    return false
+  }
+
+  if (!array1 && array2) {
+    return false
+  }
+
+  if (!array1 && !array2) {
+    return true
+  }
+
+  if (array1?.length !== array2?.length) {
+    return false
+  }
+
+  return array1
+    ? array1.every((element, index) => {
+        return array2 ? element === array2[index] : false
```

Comment:
you're assuming this as an array of a subset of primitives, isn't it?

I mean, if you compare two objects it will give false. Maybe narrowing the type would help.

So far, instead of `unknown[]` you can go with something around `type ComparablePrimitives = string | boolean | number` and `array1?: ComparablePrimitives[] | null`?

not sure at all about the names, but something around that.

Reply:
as I comment before, I think we should use this: https://lodash.com/docs/4.17.15#isEqual it's pretty well tested.

Reply:
I replaced it with lodash as @nicosampler  suggested